src/plugins/analysis/ip_and_uri_finder/test/test_ip_and_uri_finder.py

```diff
@@ -43,3 +63,26 @@ def test_process_object_uris(self):
         self.assertIn("https://www.test.de/test/", results["uris"])
         self.assertIn("ftp://ftp.is.co.za/rfc/rfc1808.txt", results["uris"])
         self.assertIn("telnet://192.0.2.16:80/", results["uris"])
+
+    def test_add_geouri_to_ip(self):
+        test_data = {'ips_v4': ['128.101.101.101', '255.255.255.255'],
+                     'ips_v6': ["1234:1234:abcd:abcd:1234:1234:abcd:abcd"],
+                     "uris": "http://www.google.de"}
+        results = self.analysis_plugin.add_geo_uri_to_ip(test_data)
+        self.assertEqual("http://www.google.de", results["uris"])
+        self.assertEqual([{"ip": "128.101.101.101", "geo_location": "(44.9759, -93.2166)"},
+                          {"ip": "255.255.255.255", "geo_location": ""}], results["ips_v4"])
+        self.assertEqual([{"ip": "1234:1234:abcd:abcd:1234:1234:abcd:abcd", "geo_location": ""}], results["ips_v6"])
+
+    @patch('geoip2.database.Reader', MockReader)
```

Comment:
if the reader is moved to the `__init__`, this patch has to be applied to all functions - including the setup.